community/opendkim/opendkim.post-install

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+selector=$(grep ^Selector /etc/opendkim/opendkim.conf | awk '{print $2}')
+keyfile=$(grep ^KeyFile /etc/opendkim/opendkim.conf | awk '{print $2}')
+domain=$(grep ^Domain /etc/opendkim/opendkim.conf | awk '{print $2}')
+
+[ "$keyfile" ] || exit 0
+
+if [ ! -f $keyfile ]; then
+	keyname=${keyfile%.private}
+	install -dm750 -g mail ${keyfile%/*}
+	openssl genrsa -out $keyfile 1024 2>/dev/null
+	openssl rsa -in $keyfile -pubout -out $keyname.public -outform PEM 2>/dev/null
+	chmod 600 $keyfile
+	chown opendkim $keyfile
+	[ "$domain" ] || exit 0
+	echo -en "$selector._domainkey IN TXT (\n\"v=DKIM1; k=rsa;\"\n\"p=" > $keyname.txt
+	grep -v ^- $keyname.public | tr -d '\n' >> $keyname.txt
+	echo -e "\"\n) ; --- DKIM key $selector for $domain" >> $keyname.txt
+	rm $keyname.public
+fi
+
+exit 0
```

Comment:
The post-install script is run right after the package is installed, i.e. _before_ user have a chance to configure it. Therefore post-install script is a not a right place for this. It’d be better to create a standalone setup script or add setup function into the init script.